### PR TITLE
feat(noir_js): Expose UltraHonk and integration tests

### LINKED ISSUE #5574: Add "honk" tagged releases of `noir_js`
### Problem

Barretenberg (UltraPlonk flavor) is facing a growing set of problems when generating proofs in JS: 
- Started with proving [_double_verify_nested_proof_](https://github.com/noir-lang/noir/issues/4409#issuecomment-2074962277)
- Recently regressed and is now failing to prove anything >2^18 constraints in JS / WASM

### Happy Case

Barretenberg (UltraHonk flavor)'s better memory footprint should help relieve the issues,

With UH now supporting recursive proofs, it has the minimum feature set to justify building and releasing "honk" tagged versions of `noir_js` for users that would like to work with it.

(That said, we should not naively flip `noir_js`'s default to UH yet as it is still [missing features](https://github.com/AztecProtocol/aztec-packages/pull/6971) to be on parity with what UP offers.)

### Project Impact

Blocker

### Impact Context

Unblocks projects that are currently blocked by proving in JS due to UP's issues.

### Would you like to submit a PR for this Issue?

None

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

## Problem\*

Resolves #5574 

## Summary\*

Similar to nargo's break from connecting to backends, we are looking to have developers use bb.js exclusively to access proving, verifying, gate counts, etc. Until the UX that NoirJS provides is worked in bb.js we can expose UltraHonk through NoirJS with minimal changes. 

This change is purely additive. I have added a new `UltraHonkBackend` that implements `Backend` and `VerifierBackend` similarly to the existing `BarretenbergBackend`. For the `Verifier` interface I just added a new method `verifyUltraHonkProof`. 

I have added integration tests that copy the existing integration tests. Once we are ready to move off of UltraPlonk the old tests and interfaces can be removed.

## Additional Context

Reference the issue for more context.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by vezenovm** [comment]
As the sol verifier is already tested within barretenberg and this work is still pre-released and not fully public, I decided to leave the UltraHonk sol verifier tests for a follow-up. The issue is [here](https://github.com/noir-lang/noir/issues/5663).

**Comment by Savio-Sou** [comment]
@noir-lang/developerrelations could you help test and review (and suggest changes if applicable) the docs on using Honk in JS? Thanks!

**Comment by critesjosh** [comment]
Sure thing. These changes are coming in 0.33?

File: tooling/noir_js_backend_barretenberg/src/backend.ts [hunk 1]
```diff
@@ -143,3 +144,140 @@ export class BarretenbergBackend implements Backend, VerifierBackend {
...
+    // There is still a proofAsFields method but we could consider getting rid of it as the proof itself
+    // is a list of field elements.
+    // UltraHonk also does not have public inputs directly prepended to the proof and they are still instead
+    // inserted at an offset.
+    // const proof = reconstructProofWithPublicInputs(proofData);
+    // const proofAsFields = (await this.api.acirProofAsFieldsUltraHonk(proof)).slice(numOfPublicInputs);
+
+    // TODO: perhaps we should put this in the init function. Need to benchmark
+    // TODO how long it takes.
+    const vkBuf = await this.api.acirWriteVkUltraHonk(this.acirUncompressedBytecode);
```
Comment (michaeljklein) [inline tooling/noir_js_backend_barretenberg/src/backend.ts:263]: I was a bit surprised to see this both in `verifyProof` and here: why not move it to init?

**Review by michaeljklein — APPROVED** [review]
LGTM (Small comment r.e. one of the TODO's)

File: tooling/noir_js_backend_barretenberg/src/backend.ts (hunk 1 above)
Comment (vezenovm) [reply tooling/noir_js_backend_barretenberg/src/backend.ts:263]: As the TODO says this needs to be benchmarked and make more modular as there are cases where we may not want a vkey in instantiate. e.g. we don't expose gate counts in this api right now, but for that we just need the circuit and don't need to generate keys. This is also leftover from the preexisting interface and this method is currently unused so I left the TODO as is.

**Pull request merged**